services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,418 @@
+package org.openkilda.functionaltests.spec.flows
+
+import static groovyx.gpars.GParsPool.withPool
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.functionaltests.extension.tags.Tag.HARDWARE
+import static org.openkilda.functionaltests.extension.tags.Tag.VIRTUAL
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.model.Cookie
+import org.openkilda.model.FlowEncapsulationType
+import org.openkilda.northbound.dto.v1.flows.PingInput
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.traffexam.TraffExamService
+import org.openkilda.testing.tools.FlowTrafficExamBuilder
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+import javax.inject.Provider
+
+@Narrative("""This spec checks basic functionality(simple flow(rules, ping, traffic, validate), pinned flow,
+flow with protected path, default flow) for a flow with VXLAN encapsulation.
+
+NOTE: A flow with the 'VXLAN' encapsulation is supported on a Noviflow switches.
+So, flow can be created on a Noviflow(src/dst/transit) switches only.""")
+class VxlanFlowSpec extends HealthCheckSpecification {
+    @Autowired
+    Provider<TraffExamService> traffExamProvider
+
+    @Unroll
+    @Tags(HARDWARE)
+    def "System allows to create/update encapsulation type for a flow\
+(#encapsulationCreate.toString() -> #encapsulationUpdate.toString())"() {
+        given: "Two active switches with traffgens"
+        def noviflowSwitchesIds = topology.activeSwitches.findAll { it.noviflow }*.dpId
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected.findAll {
+            noviflowSwitchesIds.contains(it.dpId)
+        } ?: assumeTrue("Should be at least two active traffgens connected to NoviFlow switches for test execution",
+                false)
```

Comment:
1. Find SwitchPair instead of using `def (Switch srcSwitch, Switch dstSwitch)`
2. given you have a switchpair - force selection of path that has only noviflow switches in it. Otherwise you are risking to get a non-noviflow switch as a transit one

Reply:
I don't understand why would you do this
```
def allNoviflowSwitchIds = topology.activeSwitches.findAll { it.noviflow }*.dpId
def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected.findAll {
    allNoviflowSwitchIds.contains(it.dpId)
}*.dpId
```
instead of just
```
def allTraffgenSwitchIds = topology.activeTraffGens*.switchConnected.findAll { it.noviflow }*.dpId
```